src/components/stackAggregator/index.tsx

```diff
@@ -51,275 +52,275 @@ export type StackAggregatorProps = ViewProps & {
      */
     onCollapseChanged?: (changed: boolean) => void;
     /**
-     * A setting that disables pressability on cards
+     * A setting that disables the cards' onPress
      */
     disablePresses?: boolean;
 };
 
-
-type State = {
-  collapsed: boolean;
-  firstItemHeight?: number;
-};
-
 /**
  * @description: Stack aggregator component
  * @modifiers: margin, padding
  * @example: https://github.com/wix/react-native-ui-lib/blob/master/demo/src/screens/componentScreens/StackAggregatorScreen.tsx
  */
-class StackAggregator extends PureComponent<StackAggregatorProps, State> {
-  static displayName = 'StackAggregator';
-
-  animatedScale: Animated.Value;
-  animatedOpacity: any;
-  animatedContentOpacity: any;
-  itemsCount = React.Children.count(this.props.children);
-  easeOut = Easing.bezier(0, 0, 0.58, 1);
-  animatedScaleArray: Animated.Value[];
-
-  static defaultProps = {
-    disablePresses: false,
-    collapsed: true,
-    itemBorderRadius: 0
-  };
-
-  constructor(props: StackAggregatorProps) {
-    super(props);
-    this.state = {
-      collapsed: props.collapsed,
-      firstItemHeight: undefined
-    };
-    this.animatedScale = new Animated.Value(this.state.collapsed ? buttonStartValue : 1);
-    this.animatedOpacity = new Animated.Value(this.state.collapsed ? buttonStartValue : 1);
-    this.animatedContentOpacity = new Animated.Value(this.state.collapsed ? 0 : 1);
-    this.animatedScaleArray = this.getAnimatedScales();
-  }
-
-  componentDidUpdate(_prevProps: StackAggregatorProps, prevState: State) {
-    if (prevState.collapsed !== this.state?.collapsed) {
-      LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
-    }
-  }
-
-  getAnimatedScales = () => {
-    return React.Children.map(this.props.children, (_item, index) => {
-      return new Animated.Value(this.getItemScale(index));
-    }) as Animated.Value[];
-  }
-
-  getItemScale = (index: number) => {
-    if (this.state.collapsed) {
-      if (index === this.itemsCount - 2) {
+const StackAggregator = (props: StackAggregatorProps) => {
+  const {
+    children,
+    containerStyle,
+    buttonProps,
+    collapsed = true,
+    disablePresses,
+    onItemPress,
+    contentContainerStyle,
+    itemBorderRadius,
+    onCollapseWillChange,
+    onCollapseChanged
+  } = props;
+  const itemsCount = React.Children.count(children);
+  const [firstItemHeight, setFirstItemHeight] = useState<number>();
+  const [isCollapsed, setIsCollapsed] = useState(collapsed);
+  
+  LayoutAnimation.configureNext(LayoutAnimation.Presets.easeInEaseOut);
```

Comment:
You can't trigger a LayoutAnimation like that. It will be called in each render and will also affect everything on the screen, not just the stack aggregator

Reply:
Fixed